beater/cloudbeat.go

```diff
@@ -211,7 +212,8 @@ func InitRegistry(c config.Config) (manager.FetchersRegistry, error) {
 
 // Stop stops cloudbeat.
 func (bt *cloudbeat) Stop() {
-	bt.data.Stop(bt.ctx, bt.cancel)
+	bt.cancel()
+	bt.data.Stop(bt.ctx)
```

Comment:
None of our `Stop` functions really cares about the `ctx` it receives so it has no effect but what you're doing here is calling a function with an already canceled context which would normally mean this called function won't do anything (since it's canceled).

Reply:
Maybe we should just remove the `ctx` from all the `Stop` functions.
Most fetchers don't do anything when `Stop` is called.. We just need to figure out the kube fetcher watchers stopping.

Reply:
So I removed it from `data.Stop` though the evaluator (line 217) indeed requires the context.